### DISCUSSION TITLE
Logging to file or rsyslog

### DISCUSSION THREAD

**Discussion by mntbighker** [netbox-community/netbox#5514] [Q&A] [answered]
Using the sample stanza in the docs I'm getting no logs to /var/log/netbox.log. I'm thinking permissions but so far no clues. I touched the file and chowned it netbox, still nothing. Has anyone got this to work with rsyslog? Or should it already out of the box?

**Comment by candlerb** [comment]
I presume you mean the documentation [here](https://netbox.readthedocs.io/en/stable/configuration/optional-settings/#logging) ?

Firstly, there's nothing to do with rsyslog.  This is just django writing to files.

Secondly, you will probably find that if you browse to a random URL like http://your.net.box/asdfasdf then you'll get a log like this:

```
Not Found: /asdfasdf
```

Nothing is logged when you browse pages, but this is normal. Logger `django.server` is only invoked by the development server (i.e. `manage.py runserver`), and `django.request` only logs errors.

For more info, see the Django documentation [examples](https://docs.djangoproject.com/en/3.1/topics/logging/#examples) and [loggers](https://docs.djangoproject.com/en/3.1/topics/logging/#django-s-logging-extensions), and also [stackoverflow](https://stackoverflow.com/questions/45972977/django-logging-requests):

> By default, this config sends messages from the django logger of level `INFO` or higher to the console. This is the same level as Django’s default logging config, except that the default config only displays log records when `DEBUG=True`. **Django does not log many such INFO level messages.**

> django.request will log only 4xx and 5xx requests.

-----

I tested with the following config:

```
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'file': {
            'level': 'INFO',
            'class': 'logging.FileHandler',
            'filename': '/opt/netbox/log/netbox.log',
        },
    },
    'loggers': {
        'django': {
            'handlers': ['file'],
            'level': 'INFO',
        },
    },
}
```

and I pre-created the log directory:

```
mkdir /opt/netbox/log
chown netbox:netbox /opt/netbox/log
```

On restarting netbox, the log file was created automatically (zero bytes), but only errors are logged.

**Reply by mntbighker** [reply]
> Firstly, there's nothing to do with rsyslog. This is just django writing to files.
> 

The rsyslog question was in addition to the native logging question. Presumably that's a no.

**Comment by nahun** [comment, marked as answer]
This is what I use to send logs to syslog-ng
```
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'ntb': {
            'format': 'NetBox: level=%(levelname)s,logger=%(name)s,msg=%(message)s'
        }
    },
    'handlers': {
        'syslog': {
            'level': 'INFO',
            'class': 'logging.handlers.SysLogHandler',
            'address': '/dev/log',
            'facility': 'local7',
            'formatter': 'ntb'
        }
    },
    'loggers': {
        '': {
            'handlers': ['syslog'],
            'level': 'INFO'
        }
    }
}
```

**Reply by mntbighker** [reply]
> This is what I use to send logs to syslog-ng
> 

Works like a charm, thanks my friend. For the record I use rsyslog, and it works fine for that too.
